runtime/test/FakeSimulationState.h

```diff
@@ -21,7 +21,7 @@ class FakeSimulationState : public cudaq::SimulationState {
   virtual std::unique_ptr<SimulationState>
```

Comment:
This header file and `FakeQuantumState.h` are really just scaffolding for the test and aren't included anywhere else. We could get rid of these .h files and move the code into the file `test_argument_conversion.cpp`.

Reply:
In fact, most of the code for these two classes is the same or very similar and could be shared or templated.

Reply:
done, thanks!